### PR TITLE
Start upgrade to Rails 7.0.x

### PR DESCRIPTION
**Pull request by dmarcoux**
This gives us a better idea of what needs to be done in addition to the [official guide](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html#upgrading-from-rails-6-1-to-rails-7-0).

The remaining changes needed for the Rails 7.0.x upgrade will be done in other PRs. This includes the failing tests in the `next_rails-*` checks since I am unsure about the time I can invest in this during the upcoming days.

As long as the checks for the current Rails version are passing, we can merge this PR.

### CODE REVIEW

File: .circleci/conditional_config.yml [hunk 1]
```diff
@@ -460,6 +461,7 @@ jobs:
     steps:
       - attach_workspace:
          at: .
+      - run: *install_dependencies
```
Comment (hennevogel) [inline .circleci/conditional_config.yml:464]: I guess we didn't really think about this angle when we switched to `.gem`s in git. You would need to remove the [`--local`](https://app.circleci.com/pipelines/github/openSUSE/open-build-service/7720/workflows/70fd1159-d1e0-4909-934a-796b18dce517/jobs/92677)

Comment (dmarcoux) [reply .circleci/conditional_config.yml:464]: I decided to go with caching the `.gem` files in the PR. I have also updated the [next_rails](https://github.com/openSUSE/open-build-service/wiki/next_rails/_compare/83ba6255e78d7c7adb43f41d1be1430661351c0d...0dca1095d7188cf2583db2471aec60f1398b0e0f) developer documentation to not forget this step in the future.

**Comment by dmarcoux** [comment]
I think the failures we're facing in the CI are coming from [src/api/config/initializers/wrap_parameters.rb](https://github.com/openSUSE/open-build-service/blob/e61961dfa28f33fe0ef5aef21324ed9e08fad892/src/api/config/initializers/wrap_parameters.rb).

**Comment by hennevogel** [comment]
If that would die I would be a happier men....

File: src/api/config/initializers/wrap_parameters.rb [hunk 2]
```diff
@@ -2,69 +2,40 @@
...
 
-    case request.content_mime_type
-    when Mime[:json]
-      begin
-        data = ActiveSupport::JSON.decode(request.raw_post)
-      rescue JSON::ParserError => e
-        Rails.logger.info "failed to parse JSON: #{e.message}"
-        return false
-      end
-      request.body.rewind if request.body.respond_to?(:rewind)
```
Comment (dmarcoux) [inline src/api/config/initializers/wrap_parameters.rb:53]: Rewinding the request body isn't needed anymore, this is handled in [raw_post](https://github.com/rails/rails/blob/39413de44c0e2c0dd2d964be5985b03d8f968a7b/actionpack/lib/action_dispatch/http/request.rb#L332-L341), which is what the lambdas defined in this initializer will receive when they are called.

File: src/api/config/initializers/wrap_parameters.rb [hunk 3]
```diff
@@ -2,7 +2,6 @@
 
 # This file contains settings for ActionController::ParamsWrapper which
 # is enabled by default.
-require 'active_support/core_ext/hash/conversions'
```
Comment (hennevogel) [inline src/api/config/initializers/wrap_parameters.rb:5]: The comment above still mentions `ActionDispatch::ParamsParser` which is out of the picture now right?

Comment (dmarcoux) [reply src/api/config/initializers/wrap_parameters.rb:5]: It is for `ActionController::ParamsWrapper` and it's still around. See the two `on_load` lines and their blocks below.

**Review by krauselukas — APPROVED** [review]

**Comment by krauselukas** [comment]
Will wait for the merge until the chat with @eduardoj is resolved offline :)

**Review by eduardoj — APPROVED** [review]

**Pull request merged**